services/scabbard/libscabbard/src/store/scabbard_store/diesel/operations/update_commit_entry.rs

```diff
@@ -0,0 +1,96 @@
+// Copyright 2018-2022 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use diesel::{delete, dsl::insert_into, prelude::*};
+use splinter::error::InvalidStateError;
+
+use crate::store::scabbard_store::commit::CommitEntry;
+use crate::store::scabbard_store::diesel::{
+    models::CommitEntryModel, schema::scabbard_v3_commit_history,
+};
+use crate::store::scabbard_store::ScabbardStoreError;
+
+use super::ScabbardStoreOperations;
+
+pub(in crate::store::scabbard_store::diesel) trait UpdateCommitEntryOperation {
+    fn update_commit_entry(&self, commit_entry: CommitEntry) -> Result<(), ScabbardStoreError>;
+}
+
+#[cfg(feature = "sqlite")]
+impl<'a> UpdateCommitEntryOperation for ScabbardStoreOperations<'a, SqliteConnection> {
```

Comment:
Do these need to be separate, or is can this be a single implementation with the appropriate where clause?

Reply:
I believe because of the `insert_into` they have to be separate